### PR TITLE
Adding "devDependencies" support for nuget projects.

### PR DESCRIPTION
**Pull request by mwaddell**
- Addresses issue #4659
- Supports `ItemGroup > DevelopmentDependency` groups in `.csproj` files
- Supports `developmentDependency="true"` in `packages.config`
- Does **not** support development dependencies specified using the `Condition=` property in `<Import />`, etc.

### CODE REVIEW

File: nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb [hunk 1]
```diff
@@ -109,7 +121,8 @@ def add_sdk_refs_from_sdk_tags(doc, dependency_set, project_file)
           end
         end
 
-        def build_dependency(name, req, version, prop_name, project_file)
+        # rubocop:disable Metrics/ParameterLists
+        def build_dependency(name, req, version, prop_name, project_file, dev = false)
```
Comment (feelepxyz) [inline nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb:125]: We would usually go for a keyword arg here:
```suggestion
        def build_dependency(name, req, version, prop_name, project_file, dev: false)
```

File: nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb [hunk 2]
```diff
@@ -43,6 +43,18 @@ def dependency_set(project_file:)
...
+          # Look for regular package references (development)
+          doc.css(DEV_SELECTOR).each do |dependency_node|
+            name = dependency_name(dependency_node, project_file)
+            req = dependency_requirement(dependency_node, project_file)
+            version = dependency_version(dependency_node, project_file)
+            prop_name = req_property_name(dependency_node)
+
+            dependency = build_dependency(name, req, version, prop_name, project_file, true)
+            dependency_set << dependency if dependency
+          end
```
Comment (feelepxyz) [inline nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb:56]: Could this be folded into the existing dependency selector and checking the `dependency_node` to see if it's a dev dependency?

File: nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb [hunk 3]
```diff
@@ -43,6 +43,18 @@ def dependency_set(project_file:)
...
           end
+
+          # Look for regular package references (development)
+          doc.css(DEV_SELECTOR).each do |dependency_node|
+            name = dependency_name(dependency_node, project_file)
+            req = dependency_requirement(dependency_node, project_file)
+            version = dependency_version(dependency_node, project_file)
+            prop_name = req_property_name(dependency_node)
+
+            dependency = build_dependency(name, req, version, prop_name, project_file, true)
```
Comment (feelepxyz) [inline nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb:54]:
```suggestion
            dependency = build_dependency(name, req, version, prop_name, project_file, dev: true)
```

**Review by feelepxyz — COMMENTED** [review]
Thanks for tackling this and going through all the specs! I've done a quick pass and looks good overall, wondering if the added `doc.css(DEV_SELECTOR).each` is needed or could be folder into the existing `doc.css(DEV_SELECTOR).each`?

File: nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb (hunk 2 above)
Comment (mwaddell) [reply nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb]: Yep, I just pushed that change for your review.

**Comment by mwaddell** [comment]
> Thanks for tackling this and going through all the specs! I've done a quick pass and looks good overall, wondering if the added `doc.css(DEV_SELECTOR).each` is needed or could be folder into the existing `doc.css(DEV_SELECTOR).each`?

You're welcome.  Yes, it can - I just pushed that change for your review.

**Review by feelepxyz — APPROVED** [review]
Looks great, thanks for incorporating the feedback 👍

**Pull request merged**